### PR TITLE
Add JSON loader

### LINKED ISSUE #143: Add a JSON loader

### PR DESCRIPTION
**Pull request by tlzhu19**
---

## Checklist:

<details open>

<summary></summary>

- Does this pull request close an issue? We encourage you to open an issue first if this pull request (PR) is not a
  minor change.
  - [ ] No
    - [ ] The change in this pull request is minor.
  - [X] Yes
    - Close #143

For the following questions, only check the boxes that are applicable.

- [x] Ensure that the pull request text above the horizontal line is descriptive.
- [x] Add/update relevant tests.
- [x] Add/update relevant documents.

- Do you have triage permission of this repository?
  - [ ] No
    - You are good.
  - [x] Yes
    - Does this pull request close an issue?
      - [ ] No, this pull request also acts as an issue by itself.
        - [ ] Assign yourself.
        - [ ] Label this pull request.
        - [ ] Put a milestone.
        - [ ] Put this pull request under "In Progress" or "Under Review" pipeline on ZenHub.
        - [ ] Put this pull request under the appropriate epic on ZenHub.
        - [ ] Put an estimate on ZenHub.
        - [ ] Add dependency relationship with other issues/pull requests on ZenHub.
      - [x] Yes, this pull request addresses an issue and does not act as an issue.
        - [x] Connect this pull request to the underlying issue on ZenHub.
        - [x] DON'T:
          - DON'T assign yourself or anyone else.
          - DON'T label.
          - DON'T put a milestone.
          - DON'T put this pull request under any epic on ZenHub.
          - DON'T put an estimate on ZenHub.
    - [x] When the pull request is ready, request review.
</details>

### CODE REVIEW

**Comment by xuhdev** [comment]
You also want to add a line here:

https://github.com/CODAIT/pardata/blob/57714d152f93cd64830de24312ec6e8878d175a8/pardata/_high_level.py#L253

File: pardata/loaders/_json.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from .. import typing as typing_
+from ..schema import SchemaDict
+from ._base import Loader
+
+
+class JSONLoader(Loader):
+    """JSON file to :class: list object loader."""
+
+    def load(self, path: Union[typing_.PathLike, Dict[str, str]], options: SchemaDict) -> list:
+        """The type hint says Dict, because this loader will be handling those situations in the future.
```
Comment (xuhdev) [inline pardata/loaders/_json.py]: You forgot to update the doc

File: pardata/loaders/_json.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+class JSONLoader(Loader):
+    """JSON file to :class: list object loader."""
+
+    def load(self, path: Union[typing_.PathLike, Dict[str, str]], options: SchemaDict) -> list:
+        """
+        :param path: The path to the JSON file.
+        :param options: None for JSON loader.
+        :raises TypeError: ``path`` is not a path object.
+        :return: Data loaded into a :class: list object.
```
Comment (xuhdev) [inline pardata/loaders/_json.py]: Are you sure it'll be always a list?

Comment (tlzhu19) [reply pardata/loaders/_json.py]: I wrap it with a list, otherwise it is possible to return different types

Comment (xuhdev) [reply pardata/loaders/_json.py]: It's a bit strange to force wrapping the returned value of `json.load` with a list. How about just returning `json.load`? The return type of `json.load` is `Any`: https://github.com/python/typeshed/blob/aa92ca91383fad6d85e2d9926014cd6b66fb53ee/stdlib/json/__init__.pyi#L57

File: pardata/loaders/_json.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+
+class JSONLoader(Loader):
+    """JSON file to :class: list object loader."""
+
+    def load(self, path: Union[typing_.PathLike, Dict[str, str]], options: SchemaDict) -> list:
+        """
+        :param path: The path to the JSON file.
+        :param options: None for JSON loader.
+        :raises TypeError: ``path`` is not a path object.
```
Comment (xuhdev) [inline pardata/loaders/_json.py]:
```suggestion
        :raises TypeError: ``path`` is not a path-like object.
```

File: pardata/loaders/_json.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+class JSONLoader(Loader):
+    """JSON file to :class: Any object loader."""
+
+    def load(self, path: Union[typing_.PathLike, Dict[str, str]], options: SchemaDict) -> Any:
+        """
+        :param path: The path to the JSON file.
+        :param options: None for JSON loader.
+        :raises TypeError: ``path`` is not a path-like object.
+        :return: Data loaded into a :class: Any object.
```
Comment (xuhdev) [inline pardata/loaders/_json.py]: `Any` is not a object type, it's used for representing any kind of objects in annotations. I would say:

```suggestion
        :return: An object representing loaded data. See :meth:`json.load` for details.
```

**Review by xuhdev — APPROVED** [review]

**Pull request merged**